### PR TITLE
[Maintenance] Update non-conflicting gradle dependencies

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
- Updates all but material library to latest version (read commit description)
- Cleanup build.gradle for consistency

### CODE REVIEW

File: gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationFragment.java [hunk 1]
```diff
@@ -94,9 +94,11 @@ public void onViewCreated(
     ((MainActivity) getActivity()).setActionBar(toolbar, R.drawable.ic_close_black_24dp);
     toolbar.setNavigationOnClickListener(__ -> onCloseButtonClick());
     // Observe state changes.
-    viewModel.getForm().observe(this, this::rebuildForm);
-    viewModel.getToolbarTitle().observe(this, toolbar::setTitle);
-    viewModel.getSaveResults().observe(this, e -> e.ifUnhandled(this::handleSaveResult));
+    viewModel.getForm().observe(getViewLifecycleOwner(), this::rebuildForm);
```
Comment (gino-m) [inline gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationFragment.java]: Why is this necessary? Isn't the Fragment the LifecycleOwner?

Comment (shobhitagarwal1612) [reply gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationFragment.java]: This is needed because of the updated lifecycle library: 2.1.0 --> 2.2.0

Comment (shobhitagarwal1612) [reply gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationFragment.java]: Let's revert this and handle it separately

Comment (gino-m) [reply gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationFragment.java]: I'm fine slotting it in, was just wondering why it was needed.

Comment (shobhitagarwal1612) [reply gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationFragment.java]: FYI, that wasn't breaking the assemble build, but the lint check.
https://stackoverflow.com/a/58912855

Comment (gino-m) [reply gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationFragment.java]: I see, this is very helpful, thank you!
